snap/snapcraft.yaml

```diff
@@ -35,6 +35,7 @@ apps:
       - login-session-observe
       - snap
       - snap-refresh-observe
+      - desktop-launch
```

Comment:
Is there any need for functionality granted by `desktop-launch` other than reading `.desktop` files and icons? AFAICT the `snap` plug already grants all access required for this since it is `system-files` read access to `/snap`, which I believe grants access to all subdirectories as well.

Reply:
No, it's a little bit more complex than that. The "desktop-legacy" and "unity7" plugs allow to read the folder with the .desktop files, but... they also forbid reading the contents of the .desktop files. They do this using AppArmor rules. The system-files read access is done with AppArmor rules too, and here is the problem: any deny is stronger than any allow, so it doesn't matter what you do, if you have "desktop-legacy" (which is automatically connected if you use the "desktop" plug, which snapd-desktop-integration requires to be able to show Gtk windows and notifications) or "unity7" (which we need too to be able to interact with the Dock), you won't be able to read the .desktop files contents, which we require to get the "pretty name" and the icon.

But "desktop-launch" does grant access to the contents, thanks to a patch that was merged in snapd some days ago (https://github.com/snapcore/snapd/pull/14061), because if it is connected, it will remove the lines that deny access to the .desktop files. At the beginning I did it there to avoid creating a new interface, but now that we also want to open the snap-center and the application itself after refresh, this interface is absolutely required.

Reply:
Needing it to launch the snap store and application makes good sense, thanks! I don't see `desktop-legacy` or `unity7` in the plug list, so I was confused why those deny rules would be emitted.